packages/contracts-bedrock/test/properties/PROPERTIES.md

```diff
@@ -0,0 +1,72 @@
+| 14  | supertoken total supply starts at zero                                                            | [ ]    | [ ]    |
+| 15  | deploying a supertoken registers its remote token in the factory                                  | [ ]    | [ ]    |
+| 16  | deploying an OptimismMintableERC20 registers its remote token in the factory                      | [ ]    | [ ]    |
+
+## High level
+
+| id  | description                                                                                                                                                           | halmos | medusa |
+| --- | ---                                                                                                                                                                   | ---    | ---    |
+| 17  | only calls to convert(legacy, super) can increase a supertoken’s  total supply across chains                                                                          | [ ]    | [ ]    |
+| 18  | only calls to convert(super, legacy) can decrease a supertoken’s  total supply across chains                                                                          | [ ]    | [ ]    |
+| 19  | sum of total supply across all chains is always <= to convert(legacy, super)- convert(super, legacy)                                                                  | [ ]    | [ ]    |
+| 20  | tokens sendERC20-ed on a source chain to a destination chain can be relayERC20-ed on it as long as the source chain is in the dependency set of the destination chain | [ ]    | [ ]    |
+| 21  | sum of supertoken total supply across all chains is = to convert(legacy, super)- convert(super, legacy) when all cross-chain messages are processed                   | [ ]    | [ ]    |
+
+## Atomic bridging pseudo-properties
+
+As another layer of defense, the following properties are defined which assume bridging operations to be atomic (that is, the sequencer and L2Inbox and CrossDomainMessenger contracts are fully abstracted away, `sendERC20` triggering the `relayERC20` call on the same transaction)
+It’s worth noting that these properties will not hold for a live system
+
+| id  | description                                                                                                                        | halmos | medusa  |
+| --- | ---                                                                                                                                | ---    | ---     |
+| 22  | sendERC20 decreases sender balance in source chain and increases receiver balance in destination chain exactly by the input amount | [ ]    | [x]     |
+| 23  | sendERC20 decreases total supply in source chain and increases it in destination chain exactly by the input amount                 | [ ]    | [x]     |
+| 24  | sum of supertoken total supply across all chains is always equal to convert(legacy, super)- convert(super, legacy)                 | [ ]    | [~]     |
```

Comment:
the property, from the point of view of the contracts under test, is the same one, but it will be instrumented differently (using an atomic bridge), so instead of being equal _when_ all cross chain messages are processed, it must _always_  be equal

If you believe the distinction should be an implementation detail of the testing contracts and readers of this document won't care much, I'm open to removing it

Reply:
gotcha <3